### PR TITLE
Add statsig integration

### PR DESCRIPTION
**Pull request by daniel-statsig**
## Description of the change

> Adds ability to send data to the Statsig webhook

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ]New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by arnab-p — DISMISSED** [review]

**Comment by arnab-p** [comment]
@daniel-statsig Thanks for contributing to RudderStack. Appreciate it.

As a next step, please sign the CLA using this [form](https://rudderlabs.wufoo.com/forms/rudderlabs-contributor-license-agreement)

File: v0/destinations/statsig/transform.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const {
+  defaultPostRequestConfig,
+  defaultRequestConfig
+} = require("../../util");
+const { ENDPOINT } = require("./config");
+
+function process(event) {
+  const { message, destination } = event;
```
Comment (saikumarrs) [inline v0/destinations/statsig/transform.js:8]: As I see only a few message types are supported, can you validate and throw an "unsupported message type" error?

Comment (daniel-statsig) [reply v0/destinations/statsig/transform.js:9]: Ok, I have addressed this in https://github.com/rudderlabs/rudder-transformer/pull/933/commits/003699807f61fc9a915947e9877dfa2ae080a578

Comment (saikumarrs) [reply v0/destinations/statsig/transform.js:9]: @daniel-statsig seems like you've missed importing the `CustomError` class

**Comment by daniel-statsig** [comment]
@arnab-p, I have signed the CLA. Let me know if there is anything else you need. Thanks.

File: v0/destinations/statsig/transform.js (hunk 1 above)
Comment (daniel-statsig) [reply v0/destinations/statsig/transform.js:10]: Ah, you are correct. I have fixed this add added a test case for this flow. Commit is https://github.com/rudderlabs/rudder-transformer/pull/933/commits/1428a6e2aa7e043c93e2b66a8f9e3ab953902704. Thank you.

**Review by saikumarrs — APPROVED** [review]

**Review by arnab-p — APPROVED** [review]

**Pull request merged**